### PR TITLE
Fix Pro regressions following PR #7527

### PR DESCRIPTION
**Pull request by IagoAbal**
test plan:
See returntocorp/semgrep-proprietary#664

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
CI has some problems. Can you repush an empty commit to trigger it again?

File: src/analyzing/Constant_propagation.ml [hunk 1]
```diff
@@ -134,12 +134,32 @@ let is_js env =
   | Some lang -> Lang.is_js lang
 
 let is_global = function
-  | Global -> true
-  | _ -> false
```
Comment (aryx) [inline src/analyzing/Constant_propagation.ml:138]: weird that the _else_ semgrep rule didn't find this code.

**Review by aryx — APPROVED** [review]

**Comment by IagoAbal** [comment]
Waiting for @mmcqd to confirm whether the supply-chain-related changes in the test results are OK.

**Comment by IagoAbal** [comment]
Seems to be related to PR #7506.

File: src/analyzing/Constant_propagation.ml (hunk 1 above)
Comment (IagoAbal) [reply src/analyzing/Constant_propagation.ml:138]: Yes indeed. Maybe a bug in OCaml's support? 🤔

**Pull request merged**